File: Chapter 02/JS/section_2_3_2_maintainability/after/cartItem.js
/**
 * ARCHITECTURAL NOTE: Clean File Separation
 * By moving our data structures into their own dedicated files, 
 * we make the codebase vastly easier to navigate. If another developer 
 * needs to see the shape of a CartItem, they don't have to hunt through 
 * business logic to find it.
 */
class CartItem {
    /**
     * @param {string} name 
     * @param {number} price 
     */
    constructor(name, price) {
        this.name = name;
        this.price = price;
    }
}

module.exports = CartItem;

File: Chapter 02/Python/section_2_3_2_maintainability/after/cart_item.py
"""
ARCHITECTURAL NOTE: Clean File Separation
By moving our data structures into their own dedicated files, 
we make the codebase vastly easier to navigate. If another developer 
needs to see the shape of a CartItem, they don't have to hunt through 
business logic to find it.
"""

class CartItem:
    def __init__(self, name: str, price: float):
        """
        The price of a single unit of the product.
        """
        self.name = name
        self.price = price